File: 1/1.hs
import System.IO
import Data.List.Split

main = do
  h <- openFile "input.txt" ReadMode
  contents <- hGetContents h
  let
    cals = -- Array of calories per elf
      map sum $ -- Sum each array of calories
      (map.map) to_int $ -- Convert calorie strings to ints
      map lines $ -- Split strings into arrays of calorie strings
      splitOn "\n\n" contents -- Split into array of strings
      where
        to_int = read :: String -> Int
  print (maximum cals)
  hClose h

File: 1/js/1.js
const NEWLINE = "\r\n";

const fs = require("fs");

(function () {
  const contents = fs.readFileSync("../input.txt").toString();
  const cals = contents.split(NEWLINE + NEWLINE).map(s => s.split(NEWLINE).map(x => +x).reduce((a, b) => a + b, 0));
  console.log(Math.max.apply(this, cals));
})();